test/unit/proxy_rule_test.rb

```diff
@@ -157,6 +157,46 @@ class ProxyRuleTest < ActiveSupport::TestCase
     assert backend_proxy_rule.valid?
   end
 
+  class PositionUpdateOnConcurrentDeletion < ActiveSupport::TestCase
+    disable_transactional_fixtures!
+
+    setup do
+      @provider = FactoryBot.create(:simple_provider)
+    end
+
+    attr_reader :provider
+
+    test 'position is not updated when account has been deleted' do
+      proxy = FactoryBot.create(:service, account: provider).proxy
+      proxy_rule = FactoryBot.create(:proxy_rule, proxy: proxy)
+
+      provider.delete
+
+      proxy_rule.expects(:decrement_positions_on_lower_items).never
```

Comment:
This is the downside of this approach using `#act_as_list_no_update?`.

Reply:
I think it is fine
The proxy_rule is going to be deleted anyway no?
Unless you remove the `!account` In scheduled_for_position